dojo/tools/node_js/parser.py

```diff
@@ -0,0 +1,87 @@
+import json
+# import re
+from dojo.models import Finding
+
+
+class NodeJSParser(object):
+    def __init__(self, json_output, test):
+
+        tree = self.parse_json(json_output)
+        if tree:
+            self.items = [data for data in self.get_items(tree, test)]
+        else:
+            self.items = []
+
+    def parse_json(self, json_output):
+
+        try:
+            data = json_output.read()
+            try:
+                tree = json.loads(str(data, 'utf-8'))
+            except:
+                tree = json.loads(data)
+        except:
+            raise Exception("Invalid format")
+
+        return tree
+
+    def get_items(self, tree, test):
+
+        items = {}
+        skip_nodes = ['files', 'vuln_count', 'total_count']
+        for node in tree:
+            if node not in skip_nodes:
+                for vuln_cat in tree[node]:
+                    for vuln in tree[node][vuln_cat]:
+                        item = get_item(vuln, test)
+                        unique_key = item.title
+                        items[unique_key] = item
+
+        return list(items.values())
+
+
+def get_item(item_node, test):
+
+    file_name = item_node.get('filename', '')
+    line_number = item_node.get('line', 0)
+    lines = item_node.get('lines', '')
+    path = item_node.get('path', '')
+    title = item_node.get('title', '')
+    description = item_node.get('description', '')
+    impact = 'See description'
```

Comment:
this will take up space in the DB for no information. If there is nothing more than in description I would leave it empty